aem-mock/junit5/src/main/java/io/wcm/testing/mock/aem/junit5/AemContextExtension.java

```diff
@@ -112,8 +131,10 @@ public void afterEach(ExtensionContext extensionContext) {
       aemContext.getContextPlugins().executeAfterTearDownCallback(aemContext);
 
       // Tear down {@link AemContext} after test is complete.
-      aemContext.tearDownContext();
-      AemContextStore.removeAemContext(extensionContext, extensionContext.getRequiredTestInstance());
+      if (!hasStaticContext(extensionContext.getRequiredTestClass())) {
```

Comment:
this (tearDownContext & remove) has to be done in "@BeforeAll" case as well - i assume in a "AfterAllCallback" operation. also no longer relevant for the tests in this class the teardown of the context my involve operations that otherwise lead to memory leaks when not executed.